File: src/main/kotlin/tech/diggle/apps/accomodation/elections/ElectionRepository.java
package tech.diggle.apps.accomodation.elections;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Date;

public interface ElectionRepository extends JpaRepository<Election, Long> {


    Election findElectionByElectionType(String election);

    Date findElectionByElectionDate(Date election);
}

File: src/main/kotlin/tech/diggle/apps/accomodation/election/ElectionRepository.kt
package tech.diggle.apps.accomodation.election

import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ElectionRepository : JpaRepository<Election, Long> {
    fun findElectionByDate(date: Date): Election?
    fun findByDateAndType(date: Date, type: ElectionType): Election
}